frontend/app/framework/services/localizer.service.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Squidex Headless CMS
+ *
+ * @license
+ * Copyright (c) Squidex UG (haftungsbeschränkt). All rights reserved.
+ */
+
+import { Injectable } from '@angular/core';
+
+export const LocalizerServiceServiceFactory = () => {
+    return new LocalizerService();
+};
+
+enum PipeOptions {
+    TRANSLATE = 'translate',
+    LOWER =  'lower',
+    UPPER = 'upper'
+}
+
+// TODO refactor this file, substract helper functions
+@Injectable()
+export class LocalizerService {
+    private static instance: LocalizerService;
+    public translations: Object;
+
+    constructor() {
+        if (process.env.NODE_ENV !== 'production') {
```

Comment:
This should be part of the app.component not of the service itself, because it makes it impossible to test.